core/src/main/java/com/microsoft/applicationinsights/telemetry/JsonTelemetryDataSerializer.java

```diff
@@ -292,40 +307,30 @@ private void writeToJson(String name, String value, int len) throws IOException
 
     private <T> void write(T item) throws IOException {
         if (item instanceof JsonSerializable) {
-            StringWriter stringWriter = new StringWriter();
-
-            String jsonStringToAppend = createJsonFor((JsonSerializable)item);
-            if (Strings.isNullOrEmpty(jsonStringToAppend)) {
-                return;
-            }
-
-            out.write(jsonStringToAppend);
+            writeObject((JsonSerializable) item);
         } else {
             if (WRAPPER_TYPES.contains(item.getClass()))
             {
                 out.write(String.valueOf(item));
             } else {
                 String truncatedName = truncate(String.valueOf(item), 8192);
-                String sanitizedItem = sanitizeStringForJSON(truncatedName, 8192);
                 out.write(JSON_COMMA);
-                out.write(sanitizedItem);
+                sanitizeStringForJSON(out, truncatedName, 8192);
                 out.write(JSON_COMMA);
             }
         }
     }
 
-    private <T extends JsonSerializable> String createJsonFor(T value) throws IOException {
-        StringWriter stringWriter = new StringWriter();
-        JsonTelemetryDataSerializer temp = new JsonTelemetryDataSerializer(new BufferedWriter(stringWriter));
-
-        value.serialize(temp);
-        temp.close();
-        String jsonStringToAppend = stringWriter.toString();
-        if (Strings.isNullOrEmpty(jsonStringToAppend) || JSON_EMPTY_OBJECT.equals(jsonStringToAppend)) {
-            return "";
+    private <T> void writeObject(JsonSerializable value) throws IOException {
```

Comment:
Looks like we don't need the type variable anymore

Reply:
👍